### PR TITLE
Publish to Artifactory NPM

### PR DESCRIPTION
**Pull request by felixputera**
Things done in PR:

- Replace grunt/jshint with npm/eslint
- Add TravisCI and publish script to Artifactory NPM registry

TODO:

- [X] Request for Artifactory API key for TravisCI (Apparently already configured)

@zendesk/groot

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,37 +67,20 @@ The master process understands these signals:
 Contributing
 ============
 
-git pre-commit hook
+Linting
```
Comment (sebdeckers) [inline README.md]: Can you add a Travis CI badge while we're here?

```md
[![Travis CI build status](https://travis-ci.com/zendesk/ipcluster.svg?branch=master)](https://travis-ci.com/zendesk/ipcluster)
```

File: package.json [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
-  "name": "ipcluster",
-  "version": "0.0.12",
+  "author": "Zendesk",
+  "name": "@zendesk/ipcluster",
+  "version": "1.0.0",
+  "license": "Apache-2.0",
   "repository": {
     "type": "git",
-    "url": "git@github.com:zopim/ipcluster.git"
+    "url": "git@github.com:zendesk/ipcluster.git"
```
Comment (sebdeckers) [inline package.json]: FYI the `repository` field can be abbreviated using this shorthand:

```
github:zendesk/ipcluster
```

File: lib/CommandProcessor.js [hunk 3]
```diff
@@ -98,7 +98,7 @@ p.process_data = function(chunk) {
 					break;
 				}
 			}
-			while(false);
+			while(false); // eslint-disable-line no-constant-condition
```
[the remark this answers is not in the document]
Comment (felixputera) [reply lib/CommandProcessor.js]: I don't understand this line too 😅. @timotheeg what's this line for?

File: package.json [hunk 4]
```diff
@@ -1,27 +1,30 @@
...
-    "grunt": "0.4.5",
-    "grunt-cli": "",
-    "grunt-contrib-nodeunit": "*",
-    "grunt-contrib-jshint": "*"
+    "eslint": "^6.1.0",
+    "nodeunit": "^0.11.3"
   },
   "scripts": {
-    "test": "node_modules/grunt-cli/bin/grunt jshint"
+    "test": "nodeunit test/**/*.js",
```
Comment (sebdeckers) [inline package.json]: Can we run lint during CI? I think it would help catch style nits before PR review. Maybe add `pretest: npm run lint`?

Comment (sebdeckers) [reply package.json]: Never mind, lint is running in a separate Travis stage.

**Comment by sebdeckers** [comment]
Confirmed that current master (published by this PR) is the same SHA-1 as the Git Submodule used by mediator-app-runner.

- Current master: 30e4d57ba50452282fe4b140a7cda05319c13380
- Git Submodule: ipcluster @ 30e4d57 https://github.com/zendesk/mediator-app-runner/tree/c15d6bba26df122763bab0d2ed07115009bd4258/vendor

**Pull request merged**